packages/babel-plugin-extract-export-names/test/index.test.js

```diff
@@ -2,55 +2,137 @@ const babel = require('@babel/core')
-  hi: \`Hello! \${foo.toJSON() || 'hrm'}\`,
-  abc: {
-    def: 123
-  }
-}
-export const foo3 = [1, 2, 3, { a: 'b' }]
-export const A = 'baz'
-export const [B] = [1]
-export const [C, D, E] = [a, { b: 'c' }]
-export const { F } = { foo: 'bar' }
-export const { G, H } = {}
-export { Super } from './super'
-`
-
-const transform = str => {
+const transform = value => {
   const plugin = new BabelPluginExtractExportNames()
 
-  const result = babel.transform(str, {
-    configFile: false,
-    plugins: [plugin.plugin]
-  })
+  babel.transformSync(value, {configFile: false, plugins: [plugin.plugin]})
 
-  return {
-    ...result,
-    state: plugin.state
-  }
+  return plugin.state.names
 }
 
 describe('babel-plugin-extract-export-names', () => {
-  test('adds export names to state', () => {
-    const result = transform(FIXTURE)
-
-    expect(result.state.names).toEqual([
-      'foo',
-      'foo2',
-      'foo3',
-      'A',
-      'B',
-      'C',
-      'D',
-      'E',
-      'F',
-      'G',
-      'H',
-      'Super'
+  test('should capture an empty list on an empty file', () => {
+    expect(transform('')).toEqual([])
+  })
+
+  test('should capture an empty list if nothing is exported', () => {
+    expect(transform('console.log(1)')).toEqual([])
+  })
+
+  test('should capture `const`, `let`, and `var`', () => {
+    const list = ['var', 'let', 'const']
+    const expected = ['name1', 'name2', 'name3']
+    const input = list
+      .map((d, i) => `export ${d} ${expected[i]} = ""`)
+      .join('\n')
+    expect(transform(input)).toEqual(expected)
+  })
+
+  test('should capture multiple exports with commas', () => {
+    expect(
+      transform('var name1 = "a", name2 = "b"; export var name1, name2')
+    ).toEqual(['name1', 'name2'])
+  })
+
+  test('should capture multiple exports with commas and assignments', () => {
+    expect(transform('export var name1 = "a", name2 = "b"')).toEqual([
+      'name1',
+      'name2'
+    ])
+  })
+
+  // To do: should be added.
+  test('should *not* capture a named function', () => {
+    expect(transform('export function name1() {}')).toEqual([])
+  })
+
+  // To do: should be added.
+  test('should *not* capture a class', () => {
+    expect(transform('export class name1 {}')).toEqual([])
+  })
+
+  test('should capture a list', () => {
+    expect(
+      transform('var name1 = "a", name2 = "b"; export {name1, name2}')
+    ).toEqual(['name1', 'name2'])
+  })
+
+  // To do: is this correct?
+  test('should capture renames', () => {
+    expect(
+      transform(
+        'var name1 = "a", name2 = "b"; export {name1 as foo, name2 as bar}'
+      )
+    ).toEqual(['name1', 'name2'])
+  })
+
+  test('should capture object destructuring', () => {
+    expect(transform('export var {name1} = {}')).toEqual(['name1'])
+  })
+
+  // To do: is this correct?
+  test('should capture object destructuring w/ rename', () => {
+    expect(transform('export var {name1: bar} = {}')).toEqual(['name1'])
+  })
```

Comment:
:thinking: I'm not :100: sure on this one.
I think `bar` would be expected, but I'd need to double check the spec.

Reply:
Confirmed that bar should be caught:

```js
// a.mjs
export var {name1: bar} = {}

// b.mjs
import * as all from './a.mjs'
console.log(all)
```

Yields:

```
[Module] { bar: undefined }
```